### PR TITLE
98 new cat popup

### PR DESCRIPTION
**Pull request by yishain11**
I left many commented out lines in categoryCreate because there is an issue with the modal/dialog in this page, due to the focus on the select category menu, so currently the dialog is opend from new agreement page

### CODE REVIEW

File: frontend/src/components/CategorySelect.tsx [hunk 1]
```diff
@@ -2,16 +2,23 @@ import { useTranslation } from 'react-i18next';
 import { StringBank } from 'strings';
```
Comment (avirankatz) [inline frontend/src/components/CategorySelect.tsx:2]: Could you revert the changes of this file? they seem redundant. Or am I missing something here?
Also - Nadav is going to simplify this component

File: frontend/src/types/entities.ts [hunk 2]
```diff
@@ -25,6 +25,11 @@ export interface IGroup {
   color: string;
 }
 
+export interface ModalProps {
```
Comment (avirankatz) [inline frontend/src/types/entities.ts:28]: Is this still needed?

File: frontend/src/components/Dialog/types.ts [hunk 3]
```diff
@@ -0,0 +1,7 @@
+export default interface DialogProps {
+  title: string;
+  content: string;
+  openDialogState: boolean;
+  closeFunction: any;
+  createFunction: any;
```
Comment (avirankatz) [inline frontend/src/components/Dialog/types.ts:6]:
```suggestion
  closeFunction: () => void;
  createFunction: () => void;
```

File: frontend/src/components/Dialog/style.ts [hunk 4]
```diff
@@ -0,0 +1,55 @@
+import styled from 'styled-components';
```
Comment (avirankatz) [inline frontend/src/components/Dialog/style.ts:1]: We probably don't need all of this styling (we could probably use the typography component which inherits styling from `theme.ts` for textual stuff, mui button). 
Let's conduct a joint session regarding this and prettify this together?

File: frontend/src/components/Dialog/Dialog.tsx [hunk 5]
```diff
@@ -0,0 +1,49 @@
...
+        placeholder={t(StringBank.MY_NEW_CATEGORY)}
+      />
+      <ActionsContainer>
+        <Button onClick={props.closeFunction}>Cancel</Button>
+        <DialogButton
+          onClick={() => {
+            props.createFunction(inputValue);
+          }}
+        >
+          Create
```
Comment (avirankatz) [inline frontend/src/components/Dialog/Dialog.tsx:44]: Let's add a value in `bank.ts` for this string, and use it here (same goes for close button)

File: frontend/src/pages/NewAgreement.tsx [hunk 6]
```diff
@@ -64,6 +65,34 @@ const NewAgreement: FC = () => {
     setStep(step + 1);
   }
 
+  const [openDialogState, setOpenDialogState] = useState(false);
+  const [
+    createCategoryMutationFN,
+    { data: createCategoryData, loading: newCatLoading, error: newCatError },
+  ] = useMutation(addCategoryMutation);
```
Comment (avirankatz) [inline frontend/src/pages/NewAgreement.tsx:72]: We would probably want to refetch categories after this mutation, see [group.tsx:62](https://github.com/consenz-io/webapp/blob/dev/frontend/src/contexts/group.tsx#L62) for reference

**Review by avirankatz — CHANGES_REQUESTED** [review]
@yishain11 This looks good and behaves good from the user's perspective. However I think that the amount of code could be reduced. Let's conduct a joint session and implement some improvements that are related to css cleaning?

File: frontend/src/components/CategorySelect.tsx (hunk 1 above)
Comment (yishain11) [reply frontend/src/components/CategorySelect.tsx:2]: done

File: frontend/src/types/entities.ts (hunk 2 above)
Comment (yishain11) [reply frontend/src/types/entities.ts:28]: removed

File: frontend/src/components/Dialog/types.ts (hunk 3 above)
Comment (yishain11) [reply frontend/src/components/Dialog/types.ts:6]: changed

File: frontend/src/components/Dialog/Dialog.tsx (hunk 5 above)
Comment (yishain11) [reply frontend/src/components/Dialog/Dialog.tsx:44]: I will change it to get it from a prop, since we want to use it in delete agreement too

File: frontend/src/pages/NewAgreement.tsx [hunk 7]
```diff
@@ -64,6 +65,34 @@ const NewAgreement: FC = () => {
...
+
+  const onCreateCategory = (name: string) => {
+    createCategoryMutationFN({ variables: { name, group_id: groupId } });
+    if (newCatLoading) {
+      console.log('loading');
+    }
+    if (createCategoryData) {
+      console.log('got data');
+      console.log('createCategoryData', createCategoryData);
+    }
```
[the remark this answers is not in the document]
Comment (yishain11) [reply frontend/src/pages/NewAgreement.tsx:89]: the only reason I added those is because the linter is apposing the creation of unused vars. Do you use this liner rule or should I remove it?

**Comment by avirankatz** [comment]
Changes are implemented in #110
